x-pack/plugins/security_solution/public/cases/components/case_view/index.tsx

```diff
@@ -207,27 +177,51 @@ export const CaseView = React.memo(({ caseId, subCaseId, userCanCrud }: Props) =
       {casesUi.getCaseView({
         allCasesNavigation: {
           href: formattedAllCasesLink,
-          onClick: backToAllCasesOnClick,
+          onClick: async (e) => {
+            if (e) {
+              e.preventDefault();
+            }
+            return navigateToApp(`${APP_ID}:${SecurityPageName.case}`, {
+              path: allCasesLink,
+            });
+          },
         },
         caseDetailsNavigation: {
           href: caseDetailsLink,
-          onClick: () => {
-            navigateToApp(`${APP_ID}:${SecurityPageName.case}`, {
+          onClick: async (e) => {
+            if (e) {
+              e.preventDefault();
+            }
+            return navigateToApp(`${APP_ID}:${SecurityPageName.case}`, {
```

Comment:
It might be worth moving `{APP_ID}:{SecurityPageName.case}` to a const for use here, in the tests, and in the plugin.